### PR TITLE
feat: add support for csv files to evaluator callback

### PR DESCRIPTION
**Pull request by LMMilliken**
Currently, the `train_data` and `eval_data` parameters of the `finetuner.fit` function can be provided paths to CSV files, however the `query_data` and `index_data` attributes of the `EvaluationCallback` do not. This pr adds support for CSV files to the `query_data` and `index_data` attributes of the `EvaluationCallback`.

---

- [x] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: docs/walkthrough/using-callbacks.md [hunk 1]
```diff
@@ -42,8 +42,8 @@ The evaluation callback is used to calculate performance metrics for the model b
...
+The evaluation callback is triggered at the end of each epoch, in which the model is evaluated using the `query_data` and `index_data` datasets that were provided when the callback was created. These datasets can be provided in the same way the `train_data` and `eval_data` parameters of the {meth}`~finetuner.fit` method; either as a path to a csv file, a {class}`~docarray.array.document.DocumentArray` or the name of a {class}`~docarray.array.document.DocumentArray` that has been pushed on the Jina AI Cloud. See {doc}`/walkthrough/create-training-data` for more information about how to prepare your data.
```
Comment (gmastrapas) [inline docs/walkthrough/using-callbacks.md:45]:
```suggestion
The evaluation callback is triggered at the end of each epoch, in which the model is evaluated using the `query_data` and `index_data` datasets that were provided when the callback was created. These datasets can be provided in the same way the `train_data` and `eval_data` parameters of the {meth}`~finetuner.fit` method; either as a path to a CSV file, a {class}`~docarray.array.document.DocumentArray` or the name of a {class}`~docarray.array.document.DocumentArray` that has been pushed on the Jina AI Cloud. See {doc}`/walkthrough/create-training-data` for more information about how to prepare your data.

```

File: finetuner/experiment.py [hunk 2]
```diff
@@ -143,6 +143,14 @@ def create_run(
             else None
         )
 
+        if eval_callback:
+            eval_callback.query_data - build_finetuning_dataset(
```
Comment (gmastrapas) [inline finetuner/experiment.py:147]:
```suggestion
            eval_callback.query_data = build_finetuning_dataset(
```

File: finetuner/experiment.py [hunk 3]
```diff
@@ -143,6 +143,14 @@ def create_run(
...
+                eval_callback.query_data, csv_options
+            )
+            eval_callback.index_data - build_finetuning_dataset(
```
Comment (gmastrapas) [inline finetuner/experiment.py:150]:
```suggestion
            eval_callback.index_data = build_finetuning_dataset(
```

**Comment by guenthermi** [comment]
I think we should add tests for it

**Review by gmastrapas — APPROVED** [review]

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**